python/lsst/eo/pipe/biasShiftsTask.py

```diff
@@ -0,0 +1,286 @@
+        det = exp.getDetector()
+    else: 
+        det = [amp.getName()]
+    
+    shifts = {}
+
+    for thisAmp in det:
+        oscan_rows = get_oscan_rows(exp, thisAmp)
+        if method=="derivative":
+            shifts[thisAmp.getName()] = shift_in_rows(oscan_rows, **options)
+        if method=="conv":
+            shifts[thisAmp.getName()] = scan_for_shifts(oscan_rows, **options), np.mean(oscan_rows), np.mean(exp[thisAmp.getRawDataBBox()].image.array)
+    
+    if amp is None:
+        return shifts
+    else: 
+        return shifts[amp.getName()]
+    
+def butter_highpass(cutoff, fs, order=5):
+    nyq = 0.5 * fs
+    normal_cutoff = cutoff / nyq
+    b, a = scipy.signal.butter(order, normal_cutoff, btype='high', analog=False)
+    return b, a
+
+def butter_highpass_filter(data, cutoff, fs, order=5):
+    b, a = butter_highpass(cutoff, fs, order=order)
+    y = scipy.signal.filtfilt(b, a, data)
+    return y
+
+class BiasShiftsTaskConnections(pipeBase.PipelineTaskConnections,
+                            dimensions=("instrument", "detector")):
+    raws = cT.Input(name="raw",
+                    doc="Raw input frames",
+                    storageClass="Exposure",
+                    dimensions=("instrument", "exposure", "detector"),
+                    multiple=True,
+                    deferLoad=True)
+
+    bias_shifts = cT.Output(name="bias_shifts",
+                         doc="Sudden shifts in overscan biases",
+                         storageClass="DataFrame",
+                         dimensions=("instrument", "detector"))
+                         
+
+
+class BiasShiftsTaskConfig(pipeBase.PipelineTaskConfig,
+                        pipelineConnections=BiasShiftsTaskConnections):
+    pass
+
+
+class BiasShiftsTask(pipeBase.PipelineTask):
+    ConfigClass = BiasShiftsTaskConfig
+    _DefaultName = "biasShiftsTask"
+
+    headers = ['Bias Shift (ADU)' ,'Shift center (row)', 'Shift start (row)','Shift end (row)',  \
+            'HF row noise (ADU)', 'Overscan mean (ADU)', 'Imaging mean (ADU)', 'raft', 'sensor', 'amp', 'obs_id']
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def run(self, raws):
+        bias_shifts = []
+        self.log.info("Found %d raw files", len(raws))
+
+        for i, handle in enumerate(raws):
+            exp = handle.get()
+            md = exp.getMetadata()
+            det = exp.getDetector()
+            det_name = det.getName()
+            raft, slot = det_name.split('_')
+
+            self.log.info("processing %d: %s", i, handle.dataId['exposure'])
+
+            new_shifts_dic = find_shifts_in_exposure(exp)
+            for ampname, ((shifts, hfNoise), oscanMean, imageMean) in new_shifts_dic.items():
+                for shift in shifts:
+                    bias_shifts.append((shift[0], shift[1], shift[2], shift[3],  \
+                            f'{hfNoise:.3f}',f'{oscanMean:.3f}',f'{imageMean:.3f}', raft, slot, ampname, md['OBSID']))
```

Comment:
Why are you formatting `hfNoise`, `oscanMean`, and `imageMean` as strings? They could just be floats in the final data frame and not lose any precision or risk underflowing to zero.

Reply:
All this code is copied from a script that would write out to .csv files, and writing full floats was a waste of space and potentially misleading when there were only a few digits of confidence on the measurements. I agree this is unhelpful for writing DataFrames so I will just return the float value.